src/main/java/com/contrastsecurity/maven/plugin/InstallAgentContrastMavenMojo.java

```diff
@@ -25,7 +25,19 @@ public void execute() throws MojoExecutionException {
 
         getLog().info("Agent downloaded.");
 
-        project.getProperties().setProperty("argLine", buildArgLine(project.getProperties().getProperty("argLine")));
+        String applicationName;
+        if (StringUtils.isNotBlank(appId)) {
+            applicationName = getAppName(contrast, appId);
+            retrievedAppName = applicationName;
+
+            if (StringUtils.isNotBlank(appName)) {
```

Comment:
Right above this line we're retrieving `applicationName` but on this we're checking for `appName` Hard to follow.

Reply:
@gmoore It ignores the `appName` specified by user, and pulls the `appName` from TS in case they specify both the 'appId' and 'appName'.